app/src/Controller/MainController.php

```diff
@@ -4,15 +4,21 @@
 
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\Response;
+use App\Repository\GameRepository;
 
 class MainController extends Controller {
```

Comment:
As you extends Symfony Controller (https://api.symfony.com/2.7/Symfony/Bundle/FrameworkBundle/Controller/Controller.html)

You can benefit from ControllerTrait wich is used by it (https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php)

You can do the following.

- return $this->renderView(...) instead of return new Response($this->twig->render...)
- return $this->json(...) instead of return new JsonResponse($this->serializer->serialize....)
- ...

When you extends Controller, it's useless to inject services into the constructor because the Symfony Controller already has all the container (which is heavy).

So either you extends controller (and use Trait or $this->container->...), or you don't extends it an inject services in your constructor.